lib/domain/data/user_data_holder.dart

```diff
@@ -1,4 +1,12 @@
 import 'package:flutter_dasher/common/model/user.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'user_data_holder.g.dart';
+
+@Riverpod(keepAlive: true)
+UserDataHolder userDataHolder(UserDataHolderRef ref) {
+  return UserDataHolder();
+}
 
 class UserDataHolder {
```

Comment:
Could have this just been, like a regular provide class?
```suggestion
@Riverpod(keepAlive: true)
class UserDataHolder {
```